.circleci/config.yml

```diff
@@ -1,4 +1,5 @@
-version: 2
+version: 2.1
```

Comment:
Some of these changes (like this one) are slightly preemptive. I started from a version where I was making Windows builds (unsuccessfully) and just backtracked on the explicitly windows+CI portions.

Version 2.1 is required here if we want to use the windows orb on circleCI which enables use of the windows executor.